CHANGELOG.md

```diff
@@ -2,6 +2,114 @@
+* Add selected state handling to annotation plugin ([#316](https://github.com/mapbox/mapbox-maps-android/pull/316))
+* Add API for disabling vertical/horizontal scroll gestures ([#319](https://github.com/mapbox/mapbox-maps-android/pull/319))
+* Introduce API to enable render cache feature that could bring up rendering performance improvement. ([#326](https://github.com/mapbox/mapbox-maps-android/pull/326))
+* Add `removeAnnotationManager` API. ([#330](https://github.com/mapbox/mapbox-maps-android/pull/330))
+* Improve terrain's rendering performance 
+* Set 'begin' and 'end' timestamps for StyleLoaded and MapLoaded events, so that developers could check how much time it takes to load style and map, respectively
+* Added "tile-requests-delay" and "tile-network-requests-delay" runtime source properties - an API for tile requests delay
+* Introduce MapOptions.optimizeForTerrain option that allow style rendering optimizations for terrain rendering
+* The 'text-line-height' is now data-driven property
+* MapLoaded, StyleLoaded and StyleDataLoaded events now contain begin and end timestamps reflecting the effective duration timespan
+* Implement geojson source's generateId feature. When enabled, the feature.id property will be auto assigned based on its index in the features array
+* When line lablels are inside the flip state retaining range (+/- 5 degrees around the vertical direction), the lables' flip state will be kept the same
+* Improve rendering quality of fill outlines when using render cache
+
+## Bug fixes 🐞
+* Fix scalebar doesn't refresh issue. ([#331](https://github.com/mapbox/mapbox-maps-android/pull/331))
+* Trigger nested high-level animator listener correctly. ([#335](https://github.com/mapbox/mapbox-maps-android/pull/335))
+* Make compass visible when camera was mutated while compass was disabled. ([#322](https://github.com/mapbox/mapbox-maps-android/pull/322))
+* Enable LocationComponent automatically when style loaded; fix null island location puck ([#333](https://github.com/mapbox/mapbox-maps-android/pull/333))
+* Fix crash if the belowLayerId doesn't exist on the current style ([#330](https://github.com/mapbox/mapbox-maps-android/pull/330))
+* Fix skybox cubemap sampling with metal rendering backend
+* Style pack download does not cancel pending tileset descriptor resolving
+* Fixed the excessive network usage during map browsing caused by losing of the expiration date and the etag for the cached files
+* Fix excessive network usage for delayed tile requests
+* On style pack update we reset only glyphs and only when the updated options require less glyphs than currently available and we make sure ambient cache size limit is never exceeded
+* Emit 'StyleDataLoaded' and 'SourceDataLoaded' synchronously if possible, so that developers could modify style and sources before map starts rendering style
```

Comment:
```suggestion
* Emit `StyleDataLoaded` and `SourceDataLoaded` synchronously if possible, so that developers could modify style and sources before map starts rendering style
```